docs: update the differential loading section (#37913)
angular/angular: aio/content/guide/deployment.md

## Local development in older browsers

> In Angular CLI version 10 and above, differential loading is not enabled by default for the `ng build` command.

Reviewer: Maybe?

```suggestion
Differential loading is not enabled by default for application projects that were generated with Angular CLI 10 and above.
```

Side node: this section does appear to be in the wrong part of the docs. Local development seems to be unrelated to deployment.

Reviewer: Actually it appears that there are more unrelated section to deployment. I created a seperate issue https://github.com/angular/angular/issues/39735

Author: Thank alan, I will work on the issue you mentioned.

Author: So meanwhile the main issues get prioritized, I am updating the contents here, please check.

> The `ng serve`, `ng test`, and `ng e2e` commands, however, generate a single ES2015 build which cannot run in older browsers that don't support the modules, such as IE 11.

Reviewer 2: I feel this whole paragraph needs rewording. Saying that "differential loading is not enabled" and "the commands however generate a single build" sounds confusing.

Also, is differential loading really disabled? Or is it just that the browserslist configuration is such that there is no need for differential loading.

Maybe someone from the tooling team can help word this in an accurate way :smiley:

Author: Agreed, so the thing is this is due to the browserslist configuration, but only for new projects, but if you migrate it will generate both builds. I will try to rephrase it, may be @alan-agius4 can help.
